Enterprise-CMCS/macpro-mdct-mfp
# WP Copy Closeout Statusing & Copyover

### Description

This ticket covers connecting the statusing, view mode and copyover of the Closeout. After you copyover a WP, you are now able to close out a copied Initiative.

Note: I will be doing unit test on a different branch as i want to expand it for the initiative dashboard status itself



https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/127151429/7aba1153-55e1-4a42-bdd4-1374885ac7d5


### Related ticket(s)


CMDCT-3046

---
### How to test

Because this only shows up after you copy over an already filled out WP, please follow the instructions in this PR: https://github.com/Enterprise-CMCS/macpro-mdct-mfp/pull/247

1. After the copyover is made, go to "State and Territory-Specific Initiatives", select an Initiative you had filled out before.
2. Close-out information will only enable after you fix the errors from any quarters you tracked in funding sources and/or evaluation plan 
3. Once close-out is enabled, select [Edit] button.
4. In the close out page, fill out the information until [Close out initiative] is enabled. Click it and a modal will popup
5. Click [Yes, close out initiative] and it will convert the initiative to view mode.
6. Nothing should be editable in this initiative and the status should reflect that.
7. Finish filling out WP (note that once you close out an initiative you have to add another one to make the alert conditions) and submit again
8. Go through the admin approval process 
9. Create another wp, check to make sure the previous closed out initiative does not show up in this new copy
10. If all looks good, the functionality worked!

### Important updates


N/A

---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
